src/Microsoft.DotNet.Cli.Sln.Internal/SlnFile.cs

```diff
@@ -698,17 +698,34 @@ internal void ReadLine(string line, int currentLine)
 
         internal void Write(TextWriter writer)
         {
-            foreach (DictionaryEntry e in _values)
+            if (_isMetadata)
             {
-                if (!_isMetadata)
+                foreach (DictionaryEntry e in _values)
                 {
-                    writer.Write("\t\t");
+                    if (Id != null)
+                    {
+                        writer.Write(Id + ".");
+                    }
+                    writer.WriteLine(e.Key + " = " + e.Value);
```

Comment:
A lot of this file is like this. But that's because it requires knowledge of the sln file format. I don't know what kind of comment I can add here.